### PR TITLE
[ASCII-1164] Remove Start / Stop in forwarders component

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

This PR removes `Start` / `Stop` methods in the forwarder components
Reviewing commit per commit is recommended.

#### Agent shared component
Check the agent can send metrics.
#### Metrics and logs:
Check Event forwarder:
- Create a [python custom check](https://docs.datadoghq.com/developers/custom_checks/write_agent_check/)
- Replace self.gauge('hello.world', 1) by self.database_monitoring_query_sample("{}")
- Start the Agent
- Run the Agent command stream-event-platform and check the behaviour is the same as the previous version

#### Process
- Check the process agent can send metrics.

#### Container app
- Check orchestrator can send metrics.

### CODE REVIEW

File: comp/forwarder/defaultforwarder/forwarder.go [hunk 1]
```diff
@@ -28,20 +31,32 @@ type provides struct {
...
 		return provides{
 			Comp: NoopForwarder{},
 		}
 	}
+	forwarder := NewDefaultForwarder(config, log, params.Options)
+
+	lc.Append(fx.Hook{
+		OnStart: func(context.Context) error {
+			err := forwarder.Start()
+			if ignoreLifeCycleError {
```
Comment (GustavoCaso) [inline comp/forwarder/defaultforwarder/forwarder.go:51]: In which cases we want to ignore the life cycle error?

Comment (ogaca-dd) [reply comp/forwarder/defaultforwarder/forwarder.go:51]: `forwarder.Start()` error was ignored [before](https://github.com/DataDog/datadog-agent/pull/24009/commits/7619c28b75271c75476ddd3e193901d9a915518d#diff-c363b00939bb2af8b9f63471bd173091aa6335b59371d59a2c83514c3f5e698cL272) so I keep the same behavior.

**Review by vickenty — APPROVED** [review]

**Review by robertjli — APPROVED** [review]

**Review by GustavoCaso — APPROVED** [review]
LGTM

**Pull request merged**